core/src/main/java/hudson/widgets/Widget.java

```diff
@@ -67,4 +67,20 @@ public abstract class Widget implements ExtensionPoint {
     public String getUrlName() {
         return getClass().getSimpleName();
     }
+
+    /**
+     * @return The URL of the owner of this widget relative to context path. Always ends with a trailing slash.
+     * Can be <code>null</code> for backward compatibility with widgets annotated with @Extension.
+     */
+    @CheckForNull
+    protected String getOwnerUrl() {
+        return null;
+    }
+
+    /**
+     * @return the URL relative to the context path. Always ends with a trailing '/'.
+     */
+    public String getUrl() {
+        return (getOwnerUrl() == null ? "" : getOwnerUrl()) + "widget/" + getUrlName() + '/';
```

Comment:
```suggestion
        String ownerUrl = getOwnerUrl();
        return (ownerUrl == null ? "" : ownerUrl) + "widget/" + getUrlName() + '/';
```
Would it make more sense for the default return value to be `""`?